Lib9c/Arena/ArenaSimulator.cs

```diff
@@ -1,144 +1,121 @@
+                var arenaCharacters = deadPlayers as ArenaCharacter[] ?? deadPlayers.ToArray();
+                if (arenaCharacters.Any())
                 {
-                    Result = BattleLog.Result.Lose;
-#if TEST_LOG
-                    sb.Clear();
-                    sb.Append($"{nameof(TurnNumber)}: {TurnNumber}");
-                    sb.Append($" / {nameof(WaveNumber)}: {WaveNumber}");
-                    sb.Append($" / {nameof(WaveTurn)}: {WaveTurn}");
-                    sb.Append($" / {nameof(Player)} Dead");
-                    sb.Append($" / {nameof(Result)}: {Result.ToString()}");
-                    UnityEngine.Debug.LogWarning(sb.ToString());
-#endif
+                    var (deadPlayer, result) = GetBattleResult(arenaCharacters);
+                    log.Result = result;
+                    log.Add(new ArenaDead((ArenaCharacter)deadPlayer.Clone()));
+                    log.Add(new ArenaTurnEnd((ArenaCharacter)selectedPlayer.Clone(), Turn));
                     break;
                 }
 
-                // 플레이어의 타겟(적)이 없는 경우 break.
-                if (!Player.Targets.Any())
+                if (!selectedPlayer.IsEnemy)
                 {
-                    Result = BattleLog.Result.Win;
-                    Log.clearedWaveNumber = WaveNumber;
-
-                    break;
+                    log.Add(new ArenaTurnEnd((ArenaCharacter)selectedPlayer.Clone(), Turn));
+                    Turn++;
                 }
 
-                foreach (var other in Characters)
+                foreach (var other in players)
                 {
-                    var current = Characters.GetPriority(other);
+                    var current = players.GetPriority(other);
                     var speed = current * 0.6m;
-                    Characters.UpdatePriority(other, speed);
+                    players.UpdatePriority(other, speed);
                 }
 
-                Characters.Enqueue(character, TurnPriority / character.SPD);
+                players.Enqueue(selectedPlayer, TurnPriority / selectedPlayer.SPD);
             }
 
-            Log.result = Result;
-#if TEST_LOG
-            sb.Clear();
-            sb.Append($"{nameof(TurnNumber)}: {TurnNumber}");
-            sb.Append($" / {nameof(WaveNumber)}: {WaveNumber}");
-            sb.Append($" / {nameof(WaveTurn)}: {WaveTurn}");
-            sb.Append($" / {nameof(Simulate)} End");
-            sb.Append($" / {nameof(Result)}: {Result.ToString()}");
-            UnityEngine.Debug.LogWarning(sb.ToString());
-#endif
-            return Player;
+            return log;
         }
 
-        private void Spawn()
+        private static (ArenaCharacter, ArenaLog.ArenaResult) GetBattleResult(
+            IReadOnlyCollection<ArenaCharacter> deadPlayers)
         {
-            Player.Spawn();
-            _enemyPlayer.Spawn();
-            Player.Targets.Add(_enemyPlayer);
-            _enemyPlayer.Targets.Add(Player);
+            if (deadPlayers.Count > 1)
```

Comment:
just curiously, is this possible double K.O?

Reply:
There is currently no case. I just made an exception in case a new attack type 😎